### DISCUSSION TITLE
Doctrine ORM and support for attribute mapped properties in PrestaShop

### DISCUSSION THREAD

**Discussion by gross-nvs** [PrestaShop/PrestaShop#35103] [Ideas]
Hello :wave: 

Nowadays doctrine provides an lighter way to map entities properties through attributes:
https://www.doctrine-project.org/projects/doctrine-orm/en/current/reference/basic-mapping.html#property-mapping

I failed to find any issue or PR about this, is it something is considered?
Would anybody be interested by this? (other than me obviously)

**Comment by kpodemski** [comment]
Hello @gross-nvs 

I think we could migrate annotations to attributes for the core entities unless there's a valid reason not to do that.

Let me ping @jolelievre, who is probably the best person to answer this.

PS: https://getrector.com/blog/how-to-upgrade-annotations-to-attributes

**Reply by jolelievre** [reply]
I don't see any reason not to do it indeed, this could be done in v9 since we're no on PHP 8.1 minimum

**Comment by kpodemski** [comment]
Hi @gross-nvs 

If you are motivate to do that feel free to send the pull request 👍🏻 As I mentioned, it can be done using Rector and we've already used it for some code modifications

**Comment by gross-nvs** [comment]
Thanks for your replies, I will work on this soon. 👍
